src/mito2/src/config.rs

```diff
@@ -167,13 +172,75 @@ impl MitoConfig {
             );
         }
 
-        if self.enable_experimental_write_cache {
-            ensure!(
-                !self.experimental_write_cache_path.is_empty(),
-                InvalidConfigSnafu {
-                    reason: "experimental_write_cache_path should not be empty",
-                }
-            );
+        // Sets write cache path if it is empty.
+        if self.experimental_write_cache_path.is_empty() {
+            self.experimental_write_cache_path = join_dir(data_home, "write_cache");
+        }
+
+        self.inverted_index.sanitize(data_home)?;
+
+        Ok(())
+    }
+}
+
+/// Operational mode for certain actions.
+#[derive(Debug, Serialize, Deserialize, Clone, PartialEq, Eq, Default)]
+#[serde(rename_all = "snake_case")]
+pub enum Mode {
+    /// The action is performed automatically based on internal criteria.
+    #[default]
+    Auto,
+    /// The action is explicitly disabled.
+    Disable,
+}
+
+impl Mode {
+    /// Whether the action is disabled.
+    pub fn disabled(&self) -> bool {
+        matches!(self, Mode::Disable)
+    }
+
+    /// Whether the action is automatic.
+    pub fn auto(&self) -> bool {
+        matches!(self, Mode::Auto)
+    }
+}
+
+/// Configuration options for the inverted index.
+#[serde_as]
+#[derive(Debug, Serialize, Deserialize, Clone, PartialEq, Eq)]
+#[serde(default)]
+pub struct InvertedIndexConfig {
+    /// Whether to create the index on flush: automatically or never.
+    pub create_on_flush: Mode,
+    /// Whether to create the index on compaction: automatically or never.
+    pub create_on_compaction: Mode,
+    /// Whether to apply the index on query: automatically or never.
+    pub apply_on_query: Mode,
+    /// Memory threshold for performing an external sort during index creation.
+    /// `None` means all sorting will happen in memory.
+    #[serde_as(as = "NoneAsEmptyString")]
+    pub mem_threshold_on_create: Option<ReadableSize>,
+    /// File system path to store intermediate files for external sort, defaults to `{data_home}/index_intermediate`.
+    pub intermediate_path: String,
+}
+
+impl Default for InvertedIndexConfig {
+    fn default() -> Self {
+        Self {
+            create_on_flush: Mode::Auto,
+            create_on_compaction: Mode::Auto,
+            apply_on_query: Mode::Auto,
```

Comment:
As a new feature, should we disable the index feature by default in early versions?

Reply:
In the nightly build, we'll enable this feature to expose issues during testing, and we can turn it off when it's time to release.